File: KBDTypeFront/src/dto/order/OrderCreateDto.ts
export class OrderCreateDto {
    firstName: string = '';
    lastName: string = '';
    email?: string;
    phoneNumber: string = '';
    region: string = '';
    city: string = '';
    street: string = '';
    building: string = '';
    apartment?: string;
    postalCode: string = '';
    comment?: string;
    items: {
        productId: number;
        quantity: number;
    }[] = [];

    constructor(init?: Partial<OrderCreateDto>) {
        Object.assign(this, init);
    }
}

File: KBDTypeServer/Application/DTOs/OrderDtos/OrderCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace KBDTypeServer.Application.DTOs.OrderDtos
{
    public class OrderCreateDto
    {
        // Дані користувача
        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [EmailAddress]
        public string? Email { get; set; }

        [Required]
        [Phone]
        public string PhoneNumber { get; set; }

        // Адреса доставки
        [Required]
        public string Region { get; set; }

        [Required]
        public string City { get; set; }

        [Required]
        public string Street { get; set; }

        [Required]
        public string Building { get; set; }

        public string? Apartment { get; set; }

        [Required]
        public string PostalCode { get; set; }

        // Додатково
        public string? Comment { get; set; }

        // Список товарів у замовленні
        [Required]
        [MinLength(1, ErrorMessage = "Order must contain at least one item.")]
        public List<OrderItemCreateDto> Items { get; set; }
    }
}
